### ISSUE TITLE
services.AddSingleton<IEventAggregator, EventAggregator.Blazor.EventAggregator>(); ???

### ISSUE THREAD

**Issue by roustan2** [mikoskinen/Blazor.EventAggregator#6]
Hi,
Are you sure  to register as singleton is a good way ?
In this case all clients will receive notification on their pages.

**Comment by dodyg** [comment]
Yeah this makes using `services.AddEventAggregator();` make the whole library unusable for Blazor Server Side. 

The workaround is to use this instead at `ConfigureServices`

```
services.AddScoped<EventAggregator.Blazor.IEventAggregator, EventAggregator.Blazor.EventAggregator>();
```

**Comment by AFAde** [comment]
Hi @mikoskinen,
I am sharing the same concerns as the two other developers. For Blazor Server Side, normally you would use it as a scoped service. Would it make sense to make the registration type configurable?

Thanks

**Comment by ryanbuening** [comment]
@mikoskinen, @dradovic are you accepting PR's to fix this scope issue for Blazor Server?

Also, with the current library, how can I use:

`services.AddScoped<EventAggregator.Blazor.IEventAggregator, EventAggregator.Blazor.EventAggregator>();`

and set `AutoRefresh = true`?

**Comment by gregoryagu** [comment]
It is unexpected to be a singleton, instead of scoped.  I also vote to change this. It's a super easy fix, no reason not to.
